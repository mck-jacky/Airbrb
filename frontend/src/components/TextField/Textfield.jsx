import React from 'react'
import PropTypes from 'prop-types';
import TextField from '@mui/material/TextField';

const Textfield = ({ id, label, name, autoFocus, type }) => {
  return (
    <TextField data-testid='textfield'
      margin="normal"
      required
      fullWidth
      id={id}
      label={label}
      name={name}
      inputProps={{ 'aria-label': { name } }}
      autoFocus={autoFocus}
      type={type}
    />
  )
}

Textfield.propTypes = {
  id: PropTypes.string,
  label: PropTypes.string,
  name: PropTypes.string,
  autoFocus: PropTypes.bool,
  type: PropTypes.string,
}

export default Textfield
