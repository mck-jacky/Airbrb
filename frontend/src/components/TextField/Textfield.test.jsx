import React from 'react';
import { render, screen } from '@testing-library/react';
import Textfield from './Textfield';
import ShallowRenderer from 'react-test-renderer/shallow'

describe('Textfield', () => {
  it('is in the document', () => {
    render(<Textfield id='login-password' label='password' name='password' />)
    const component = screen.getByTestId('textfield')
    expect(component).toBeInTheDocument();
  })

  it('has text', function () {
    render(<Textfield id='login-password' label='password' name='password' />)
    const component = screen.getByTestId('textfield')
    expect(component).toHaveTextContent('password')
  });

  it('has no initial value', function () {
    const { container } = render(<Textfield id='login-password' label='password' name='password' />)
    const input = container.querySelector('input')
    expect(input).toHaveValue('')
  })

  it('has to match the snapshot', () => {
    const renderer = ShallowRenderer.createRenderer()
    renderer.render(<Textfield id='login-password' label='password' name='password' />)
    const result = renderer.getRenderOutput()
    expect(result).toMatchSnapshot()
  })
})
