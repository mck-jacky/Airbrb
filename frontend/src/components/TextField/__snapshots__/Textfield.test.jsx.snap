// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Textfield has to match the snapshot 1`] = `
<ForwardRef(TextField)
  data-testid="textfield"
  fullWidth={true}
  id="login-password"
  inputProps={
    Object {
      "aria-label": Object {
        "name": "password",
      },
    }
  }
  label="password"
  margin="normal"
  name="password"
  required={true}
/>
`;
